ibis/expr/types/relations.py

```diff
@@ -1177,6 +1181,30 @@ def prevent_rewrite(self, client=None) -> TableExpr:
     def materialize(self) -> TableExpr:
         return self
 
+    def alias(self, alias: str) -> ir.TableExpr:
+        import ibis.expr.operations as ops
+
+        expr = ops.View(
+            child=self,
+            name=alias,
+            backend=self._find_backend(),
+        ).to_expr()
+
+        # NB: calling compile is necessary so that any temporary views are
+        # created so that we can infer the schema without executing the entire
+        # query
+        expr.compile()
```

Comment:
I don't follow, can you point me where views get created?

Reply:
https://github.com/ibis-project/ibis/pull/3642/files#diff-9f928791a077be8718dc80822ee8d11db44bf4f1fac268e6897232ed46255d99R105